nixos/modules/config/supported-hashes.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
```

Comment:
This script could be simpler with a `bash` script using regex. It can also avoid pulling `python` dependency to the activation script.

```bash
users=()
while IFS=: read -r user hash tail; do
    if [[ "$hash" != '!' && ! "$hash" =~ ^\$(y|gy|7|2b|2y|2a|6)\$ ]]; then
        users+=("$user")
    fi
done </etc/shadow
```

Reply:
Thanks!